ethcore/client-traits/src/lib.rs

```diff
@@ -231,12 +236,14 @@ pub trait BlockChainClient : Sync + Send + AccountData + BlockChain + CallContra
 	fn block_hash(&self, id: BlockId) -> Option<H256>;
 
 	/// Get address code at given block's state.
-	fn code(&self, address: &Address, state: StateOrBlock) -> Option<Option<Bytes>>;
+	fn code(&self, address: &Address, state: StateOrBlock) -> StateResult<Option<Bytes>>;
```

Comment:
`StateResult` seems to re-inventing the `Result type` why not `Result<Bytes, &'static str>` instead? Or is there any particular reason why `StateResult` is needed?

Reply:
It would be `Result, Option<Bytes>, something>` which doesn't really help understanding much. The name `StateResult` does make me think of `Result` but I'm not sure what a better name is. 
The point of it is, if I understand the issue correctly, to distinguish and communicate three possible states: "all good, here's the code", "all good, but there is no code there", and "not ok, there should have been code there but wasn't"

Reply:
We could have a type alias like `type StateResult = Result<Option<Bytes>, StateError>; type StateError = ();`, but at this point it seems like bikeshedding.

Reply:
yeah alright, thanks. Let's keep it as it is then.